docs/rules/quote-props.md

```diff
@@ -136,6 +140,75 @@ var object3 = {
 };
 ```
 
+#### consistent
+
+When configured with `"consistent"`, the patterns below are considered warnings. Basically `"consistent"` means all or no properties are expected to be quoted, in other words quoting style can't be mixed within an object. Please note the latter situation (no quotation at all) isn't always possible as some property names require quoting.
+
+```js
+var object1 = {
+    foo: "bar",
+    "baz": 42,
+    "qux-lorem": true
+};
+
+var object2 = {
+    'foo': 'bar',
+    baz: 42
+};
+```
+
+The following patterns are considered okay and do not cause warnings:
+
+```js
+var object1 = {
+    "foo": "bar",
+    "baz": 42,
+    "qux-lorem": true
+};
+
+var object2 = {
+    'foo': 'bar',
+    'baz': 42
+};
+
+var object3 = {
+    foo: 'bar',
+    baz: 42
+};
+```
+
+#### always-if-needed
```

Comment:
Too confusing, as you even day this behaves like consistent. Please change to `consistent-as-needed`.